File: backend/src/main/java/com/urmyfood/backend/domain/repository/OrderRepository.java
package com.urmyfood.backend.domain.repository;

import com.urmyfood.backend.domain.model.Order;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface OrderRepository {
    Order save(Order order);
    List<Order> findByCustomerId(Long customerId);
    List<Order> findByShopId(Long shopId);
    List<Order> findShopOrdersForStatistics(Long shopId, OffsetDateTime startAt, OffsetDateTime endAt);
    List<Order> findShopOrdersForAllTimeStatistics(Long shopId);
    List<Order> findPendingOrAcceptedOrdersByPostId(UUID postId);
    List<Order> findPendingOrAcceptedOrdersByShopId(Long shopId);
    Optional<Order> findById(UUID orderId);
    Optional<Order> findByIdForUpdate(UUID orderId);
    Optional<Order> findByPayosOrderCode(Long payosOrderCode);
    List<Order> findPendingExpiredOrders(OffsetDateTime expiredBefore);
    List<Order> findAcceptedUnpaidExpiredOrders(OffsetDateTime expiredBefore);
}

File: frontend/app-shop/src/main/java/com/urmyfood/shop/domain/repository/OrderRepository.kt
package com.urmyfood.shop.domain.repository

import com.urmyfood.shared.domain.model.Result
import com.urmyfood.shop.domain.model.Order

interface OrderRepository {
    suspend fun getShopOrders(token: String): Result<List<Order>>
    suspend fun getShopOrderDetail(token: String, orderId: String): Result<Order>
    suspend fun updateOrderStatus(token: String, orderId: String, status: String, rejectReason: String?): Result<Order>
}
